ui/controlgroup.js

```diff
@@ -0,0 +1,175 @@
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map(function(){
+				return $( this )[ widget ]( "widget" )[ 0 ];
+			}).removeData( "ui-controlgroup-data" );
+		});
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			var options = {};
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+			if ( $.fn[ widget ] && selector ) {
+				that.element
+					.children( selector )[ widget ]( method ? method: options )
+					.each(function(){
+						if ( method !== "destroy" ) {
+							$( this )[ widget ]( "widget" ).data( "ui-controlgroup-data", {
+								"type": widget,
+								"element": $( this )
+							});
+						}
+					});
+			}
+		});
+	},
+
+	_button_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-button": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_checkboxradio_options: function( position ) {
+		var cornerClasses = {
+			"middle": null,
+			"first": "ui-corner-" + ( ( this.options.direction === "vertical" )? "top" : "left" ),
+			"last": "ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom" : "right" )
+		};
+
+		return {
+					classes: {
+						"ui-checkboxradio-label": cornerClasses[ position ]
+					}
+				};
+	},
+
+	_selectmenu_options: function( position ) {
+		var classes = {
+			middle: {
+				"ui-selectmenu-button-open": null,
+				"ui-selectmenu-button-closed": null
+			},
+			first: {
+				"ui-selectmenu-button-open":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "tl" ),
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "top": "left" )
+			},
+			last: {
+				"ui-selectmenu-button-open":
+					( this.options.direction === "vertical" )? null: "ui-corner-tr",
+				"ui-selectmenu-button-closed":
+					"ui-corner-" + ( ( this.options.direction === "vertical" )? "bottom": "right" )
+			}
+
+		};
+		return { classes: classes[ position ] };
+	},
+
+	_setOption: function( key, value ) {
+		var original = this.options[ key ];
+
+		this._super( key, value );
+		if ( key === "direction" ) {
+			this.element.removeClass( "ui-controlgroup-" + original );
+		}
+		if ( key === "disabled" ) {
+			this._callChildMethod( value ? "disable" : "enable" );
+		} else {
+			this.refresh();
+		}
+
+	},
+
+	refresh: function() {
+		var children,
+			that = this;
+
+		this._addClass( "ui-controlgroup ui-controlgroup-" + this.options.direction );
+		this._callChildMethod();
+		children = this.element.children( ".ui-button" );
```

Comment:
So the use of `children` here came up in adopting in mobile and @jzaefferer had previously questioned this too. Right now this will fail if a checkboxradio has its input nested inside the label because the input itself is not a child. This can be fixed by simply changing chidlren call to find and in the items option prefix all the selectors with `>` and for checkboxradios adding `>label>input[type='radio']`. Personally I think we should stick this this only being children i dont think we want to support arbitrary wrapping of elements thats entirely likely to break formating. If people want to adjust this behavior its easily don't through the use of the items option. The `ui-button` will likely still need to change at some point as we support non button types, but I think we should keep this to just being children just do it via selector rather then method.

Ill bring this up in the meeting today.